src/common/string/compare.ts

```diff
@@ -11,3 +11,24 @@ export const compare = (a: string, b: string) => {
 
 export const caseInsensitiveCompare = (a: string, b: string) =>
   compare(a.toLowerCase(), b.toLowerCase());
+
+export const fuzzyFilter = (filter: string, word: string) => {
```

Comment:
Went with "fuzzyFilter" instead of "fuzzyCompare" here because it's not _really_ comparing two strings. It's deciding whether an entity should be _filtered out_ of a list.

Reply:
We use fuse.js for Fuzzy searching <https://fusejs.io/>

Reply:
I might be missing an option, but I don't think fuse supports this "fuzzy sequence matching" thing I'm proposing here. 

What I'm proposing is something like Sublime/VSCode's "quick open" file filter. Where you can type out bits and pieces of a word rather than write a word that's similar to the target. 

For example, "senlef" should match "**sen**sor.bed\_sensor\_**t**op\_**lef**t".